packages/stylelint-plugin/src/rules/no-invalid-equations.ts

```diff
@@ -0,0 +1,165 @@
+import { Declaration, Root } from "postcss"
+import postcssValueParser, { WordNode } from "postcss-value-parser"
+import {
+  kaizenVariableUsedNextToOperatorMessage,
+  negatedKaizenVariableMessage,
+  tokenNotInterpolatedInCalcMessage,
+} from "../messages"
+import { isOperator } from "../patterns"
+import { Options } from "../types"
+import { isVariable, parseVariable, stringifyVariable } from "../variableUtils"
+import { walkDeclsWithKaizenTokens, walkWithParent } from "../walkers"
+
+export const noInvalidEquationsRuleName = "no-invalid-equations"
+export const noInvalidEquationsOnDeclaraion = (
+  postcssNode: Declaration,
+  parsedValue: postcssValueParser.ParsedValue,
+  options: Options
+) => {
+  /*
+    This rule will attempt to detect whether a Kaizen token is used as a term within an equation.
+    It does this by detecting whether an operator and a token are next to each other in a significant way.
+
+    `lastTwoWordNodesNextToEachOther` is used to facilitate this, by looping through the value and analysing each pair of significant adjacent word nodes.
+    Adjacency is determined to be true if they aren't separated by a div (comma), or a function.
+
+  */
+
+  const parsed = parsedValue
+  let amended: string | undefined
+
+  // Will use this to detect whether two invalid word nodes are used next to each other
+  // Will reset when the parent changes
+  let lastTwoWordNodesNextToEachOther: WordNode[] = []
+  walkWithParent(parsed, (node, parent) => {
+    if (node.type === "function" || node.type === "div") {
+      // Reset the word sequence if there is something meaningful separating them, like a function or a divider (comma)
+      lastTwoWordNodesNextToEachOther = []
+      return
+    }
+
+    const isWithinCalcFunction = parent?.value === "calc"
+    if (node.type === "word") {
+      // Detect the cose of the node being a negated variable (because postcss-value-parser doesn't detect the operator and variable as seprate nodes)
```

Comment:
```suggestion
      // Detect the case of the node being a negated variable (because postcss-value-parser doesn't detect the operator and variable as seprate nodes)
```